website/docs/ref/react.doc.js

```diff
@@ -197,6 +197,28 @@ function renderButton(container: HTMLElement, visited?: boolean) {
 
   Any function that returns a React element can be used as a component class in
   a JSX expression.
+
+*/
+const AsyncGreeter = loadAsync(Greeter, new Promise((resolve, reject) => {
```

Comment:
The weird thing is:
1. The `ReactClass` types don't work in practice. They don't actually infer types correctly.
2. If I try to make my own type definition for both classes and functions the HOC types don't work.
3. So, I HAVE to use overloads to support both classes and functions.

But then functions may or may not have defaultProps. And again, if I try to combine both into one type, the types don't quite flow correctly. This means that there needs to be one MORE overload.

4. So with three function overloads for a single HOC, the types get quite hairy. This is why I started with just the case for class.